packages/flutter/lib/src/animation/animation.dart

```diff
@@ -230,6 +233,32 @@ abstract class Animation<T> extends Listenable implements ValueListenable<T> {
     return child.animate(this as Animation<double>);
   }
 
+  /// Create a new [Animation] that transforms this animation using the provided
```

Comment:
In theory, you can achieve the same thing by implementing a custom `Animatable` class (only have to implement the `transform` method with the closure you would provide here) and then use `Animation.drive`.

This API avoids the extra class definition, I guess. Is that really such a hassle?

Reply:
Oh, and I guess Animation.drive only works if the parent is a `Animation<double>`. This one in theory allows any Animation to be transformed into any other... Is that a feature you need?

Reply:
Ahh maybe, let me look into that more.